src/core/email-client.js

```diff
@@ -53,102 +72,241 @@ function getFreshImap (cb) {
 }
 
 class EmailClient {
-    constructor () {
+    //constructor takes the full name of user as String
+    constructor (name, eventBus) {
         var self = this;
-
-        //  Get Imap obj and check emails
-        getFreshImap(function (err, imap) {
+        self._name = name; 
+        self._email = "";
+        self._imap = Q.defer();
+        self._transporter = Q.defer();
+        self._status = Q.defer();
+        self._SendCheck = Q.defer();
+        self._changeCheck = Q.defer();
+        self._eventBus = eventBus;
+       
+        getFreshImap(function (err, imap, email, smtpConfig) {
             if (err) {
                 console.log(err);
             } else {
-                self.imap = imap;
-                self._checkEmails();
+                self._imap.resolve(imap);
+                self._email = email;
+                self._transporter.resolve(NodeMailer.createTransport(smtpConfig));
             }
         });
     }
 
-    _checkEmails () {
+    _loadInfo(){
+    //  Get Imap obj and check emails
+        var self = this;
+        return self._imap.promise;
+    }
+    //get transporter promise
+    _loadTransporter(){
+        var self = this; 
+        return self._transporter.promise;
+    }
+    //get status promise
+    _loadStatus(){
+        var self = this;
+        return self._status.promise;
+    }
+    //get sendcheck promise
+    _loadSendCheck(){
+        var self = this;
+        return self._SendCheck.promise;
+    }
+    //get changeCheck promise
```

Comment:
I know I'm being strict about this but our method comments need to be standardized, so I'd like for you to follow the format you see in model.js.
In addition, please be a little more descriptive. We should be able to know what things like the transporter and sendCheck are for without having to look it up.